### PR TITLE
README reworked

### PR DESCRIPTION
**Pull request by comradekingu**
Mentioning Hosted Weblate is one of the requirements in https://hosted.weblate.org/hosting/
It doesn't have to be all the widgets, but it is nice to see what languages are available.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,30 +110,30 @@ To clone a project, use the command:
...
 include ':tordnscrypt', ':filepicker'
 project(':filepicker').projectDir = new File('android-filepicker/filepicker')
-//Please comment line below if you are not the project owner
+//Please comment out the line below if you are not the project owner
 //project(':tordnscrypt').buildFileName = 'owner.gradle'
 ```
 
-It is expected that you can solve build problems yourself, so there is no support for building. 
-If you cannot build yourself, there are prebuilt versions of **InviZible Pro** available [here](https://github.com/Gedsh/InviZible/releases/latest).
+As is expected, if you can solve how to build problems for yourself, so there is no support for how you got there. :)
```
Comment (Gedsh) [inline README.md:127]: What is it?

File: README.md [hunk 2]
```diff
@@ -109,30 +110,30 @@ To clone a project, use the command:
...
 #### Translating
 
-If you would like to translate this application into other languages,
-please send an email to the developer [invizible.soft@gmail.com](mailto:invizible.soft@gmail.com)
+[Translate InviZible on Hosted Weblate](https://hosted.weblate.org/engage/invizible/).
+
+[![Translation status](https://hosted.weblate.org/widgets/invizible/-/multi-auto.svg)](https://hosted.weblate.org/engage/invizible/?utm_source=widget)
 
-Thanks to [POEditor](https://poeditor.com/), you can translate InviZible Pro into your language with pleasure and for free.
-Just follow this [link](https://poeditor.com/join/project/h6ulNL9gEd).
```
Comment (Gedsh) [inline README.md:135]: Why have you removed POEditor? InviZible continues to support translation with it.

File: README.md [hunk 3]
```diff
@@ -166,9 +167,8 @@ and carries no guarantee from The Above Projects about quality, suitability or a
 
 [GNU General Public License version 3](https://www.gnu.org/licenses/gpl-3.0.txt)
 
-Copyright (c) 2019-2021 Garmatin Oleksandr invizible.soft@gmail.com
-
-All rights reserved
+Copyright © 2019–2021 Garmatin Oleksandr and contributors
+invizible.soft@gmail.com
```
Comment (Gedsh) [inline README.md:171]: Please do not change the copyright. If you disagree with it, unfortunately I cannot accept the merging of your pull requests.

File: README.md [hunk 4]
```diff
@@ -26,60 +27,60 @@ They are used to achieve maximum security, privacy and comfortable use of the In
 * Can block "adult" sites *
 * Hides visited sites from your provider
 * Prevents some types of resource blocks
-* Open-source
+* Libre software
```
Comment (Gedsh) [inline README.md:30]: Does everyone really know what libre software is?

File: README.md (hunk 3 above)
Comment (comradekingu) [reply README.md:171]: You need a CLA to have sole copyright and that was never the case. "and contributors" is accurate.

File: README.md (hunk 4 above)
Comment (comradekingu) [reply README.md:30]: No, but it is all there in the word (as opposed to "open source".

File: README.md (hunk 1 above)
Comment (comradekingu) [reply README.md:127]: Same joke I thought you were going for.

File: README.md (hunk 2 above)
Comment (comradekingu) [reply README.md:135]: POEditor is not a good tool, and advertising it isn't something I want to do.

File: README.md [hunk 5]
```diff
@@ -2,18 +2,19 @@
...
-
-### [Download the latest version from Github](https://github.com/Gedsh/InviZible/releases/latest)
+[![Translation status](https://hosted.weblate.org/widgets/invizible/-/invizible/svg-badge.svg)](https://hosted.weblate.org/engage/invizible/?utm_source=widget)
 
 ### [IzzyOnDroid F-Droid beta version](https://apt.izzysoft.de/fdroid/index/apk/pan.alexander.tordnscrypt)
 
 ### [F-Droid stable version](https://f-droid.org/packages/pan.alexander.tordnscrypt.stable/)
 
-## Android application for Internet privacy and security
+### [Google Play stable version](https://play.google.com/store/apps/details?id=pan.alexander.tordnscrypt.gp)
```
[the remark this answers is not in the document]
Comment (comradekingu) [reply README.md:11]: I moved Google Play to the bottom since it is the worst option.

[the remark this answers is not in the document]
Comment (Gedsh) [reply README.md:11]: 99% of the project support is from Google Play. Beginners usually start with it. InviZible is intended not only for geeks, but for all users, even if they are just starting their journey to privacy and anonymity.

File: README.md (hunk 1 above)
Comment (Gedsh) [reply README.md:127]: Not funny, as for me. I often suggest that beginners read my github.

File: README.md (hunk 2 above)
Comment (Gedsh) [reply README.md:135]: POEditor is a simple tool for anyone to translate. Webplate is very complex.

File: README.md (hunk 3 above)
Comment (Gedsh) [reply README.md:171]: This is my project anyway. So if you disagree, I can only suggest that you fork and develop your own application however you want.
I see no point in cooperation. You wanted to translate a project, but you really want to redo it. This is unacceptable to me. Sorry if this was not clear from the beginning.

Comment (comradekingu) [reply README.md:171]: I have zero changes in your project. I am adding that as an acknowledgement of others.
Your project includes code from different projects, and has accepted contributions since then.
How you declare copyright has no bearing on what is covered by their copyright, because that is not how copyright works.
Just trying to help.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (comradekingu) [reply README.md:11]: F-Droid is not intended for geeks. If users could be moved to other platforms then that would be great.
Especially concerning matters "privacy and anonymity".
If people aren't asking about the specifics of Google Play, then F-Droid and Izzy isn't a worse option.
If they are asking about that, it is more reason to move it down the queue.

File: README.md (hunk 1 above)
Comment (comradekingu) [reply README.md:127]: Will change.

Comment (comradekingu) [reply README.md:127]:
```suggestion
There is no support offered for building the app yourself, as this is an advanced option.
```

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Gedsh) [reply README.md:11]: Users are smart enough to make their own choices. And this is not related to the translation of the project, is it?

File: README.md (hunk 2 above)
Comment (comradekingu) [reply README.md:135]: And they all do, except there isn't any completion of languages, and there are very few translators.
There is no community there. People use InviZible first and then use whatever tool it mandates second.

POEditor is a simplistic tool that doesn't provide enough utility to fix the errors it produces.
https://hosted.weblate.org/checks/?project=invizible&component=invizible

Weblate has users, a libre license, it is the better tool, and it produces better results.
As a service, Hosted Weblate has better terms
>CODE WHALE reserves the right to change the Terms from time to time in its sole discretion without any notifications regarding this matter. 

There are no libre software projects using POEditor that I know of. Why should InviZible use or advertise it?

File: README.md (hunk 3 above)
Comment (Gedsh) [reply README.md:171]: Actually, the problem is not with the copyright. The problem is you are making changes that are not translation related. I appreciate your help, but your changes are not acceptable. I accept all contributors, but I accept only those changes that, in my opinion, are useful for the project. So you cannot force me to merge your changes.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (comradekingu) [reply README.md:11]: Great. I am just trying to list the better options for users first. It is consistent with other projects, which is a feature of localization. Providing something users can derive meaning from and get some level of familiarity from is not irrelevant.

File: README.md (hunk 3 above)
Comment (comradekingu) [reply README.md:171]: This is very useful, because it shows gratitude for the help you got along the way.
I am not trying to force you to do anything.
I make changes to work I have done if the changes make sense.
You seem to have an idea about copyright or my intentions that isn't shared by me.

If I spend the effort to fix the language, it really makes sense for me to also improve the landing page.
That is where my honor lies.

File: README.md (hunk 2 above)
Comment (Gedsh) [reply README.md:135]: > Weblate has users, a libre license, it is the better tool, and it produces better results.

InviZible has been translated into 10 languages without any problem using POEditor. And I have a problem fixing only English translation with Webplate. The result is obvious.

> Why should InviZible use or advertise it?

Because it produces result and a free plan for open source projects. So why not.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Gedsh) [reply README.md:11]: Do you think users are not smart enough to make their own choices? Or maybe you think that Play Market users shouldn't use InviZible, since they don't even need a piece of privacy?

File: README.md (hunk 3 above)
Comment (Gedsh) [reply README.md:171]: I can accept your help if you only do the translation. If you disagree, I cannot merge both pool requests. I appreciate the help of the contributors. It is very important for me. But everyone must do their own thing, otherwise the project has no future.

File: README.md (hunk 2 above)
Comment (comradekingu) [reply README.md:135]: The 10 languages are full of errors, some of which are incomplete.
There is a nb_NO translation made on Weblate, and it isn't good.
First the source has to make sense, then the translations can make sense.
Sending people to the place that cares for user freedom and happens to be the better software is less complicated and better for translation-quality.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (comradekingu) [reply README.md:11]: I think the Play store is way worse for privacy, I make no observations about users.
There is a preference in mentioning the better alternatives first, and that is not the Play store.
To mention the Play-store lasts fits into the strategy of paying tribute to software that is on the side of apps like this one.
People like that, and are more likely to donate because of it.

If I am somehow wrong about it, it can be changed. So far that has happened in no projects, and I have re-ordered it in many.

File: README.md (hunk 2 above)
Comment (Gedsh) [reply README.md:135]: I give people the choice to use POEditor or Webplate. For me, choice is freedom. You decided instead of everyone. Is this such modern freedom?

File: README.md (hunk 3 above)
Comment (comradekingu) [reply README.md:171]: One PR has nothing to do with the other.
Most importantly mentioning Hosted Weblate is one of the terms of using that platform. Other than that I don't understand what "translation" is supposed to mean.
I offered to set up Weblate, translate some languages, fix up the source strings, and set up donations.
None of that is for me, all of it is free labour for you.
What I do is this sort of thing. I can't imagine how that isn't a fantastic deal for you, seeing as you don't want to spend the time doing it.
I am like that too, I spend my time fixing things, not redoing unimportant stuff.

You can do with this README PR exactly what you want, subject to the terms of the license.
The translation I have already made is subject to the same copyright, and that is shared on the terms of the license.
If having contributors is important to you, appreciating them by mentioning it in the copyright notice is a good gesture.
It changes absolutely nothing other than to send some goodwill out into the world.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Gedsh) [reply README.md:11]: You talk about dreams. The reality is different. 99% of donations come from the Play Market. And I respect those people who need the Play Market, but don't forget to support open source projects.

File: README.md (hunk 2 above)
Comment (comradekingu) [reply README.md:135]: The freedom of choice isn't removed. I decide nothing for anyone. That is your all your decision.

Applying the same level of consistency to the app would be to argue for a setting to decrypt all traffic too, albeit at a logical extreme.

Similarly, a website that lists your app next to some shady VPN service, or shoes for sale, isn't promoting more freedom, but less, and with more noise.
To say entropy is tantamount to freedom amounts to practical sophistry, and the amount of available and good choices dwindle from making poor judgements.
Nobody actually wants to use POEditor, yet. No libre software projects actually use it that I know of.
It is neither simple, its results are not good, and the feature is not in demand.

I try not to mislead libre software translators into a only-as-a-service platform that can change their terms whenever they want. It also loads Google Analytics and a bunch of other stuff.
Not everyone knows about that, so using the good name of InviZible to promote a service like that just doesn't serve the project. It may seem like a comparable product when it isn't.

POEditor is the opposite of freedom, and there is not less of it in any meaningful sense if the option is not available, or not advertised.

TL;DR All mentioning POEditor amounts to is to undermine the principles of the app and the better translation platform.

File: README.md (hunk 3 above)
Comment (Gedsh) [reply README.md:171]: > One PR has nothing to do with the other.

My copyright in every InviZible's file. If you disagree with it, how can I use your work? It's not fair. I want to be fair.

> Most importantly mentioning Hosted Weblate is one of the terms of using that platform

Of course, I will mention it. Unfortunately, I didn't have time for this before.

> I offered to set up Weblate, translate some languages, fix up the source strings, and set up donations.

And I appreciate your work and your time. But if you did just that, there would be no questions.

> The translation I have already made is subject to the same copyright

Therefore, you can agree to my terms or not. So I can use your incredible work or not. The choice is simple.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (comradekingu) [reply README.md:11]: I am saying you deserve more of those.
Google prevents asking for donations in app store listings.
Google taking a 30% cut of sales is a good reason to list it last.
If you make a Liberapay account I can help you promote it and write a pitch for why people should donate.
Given the option to use all three stores and all donation platforms, people can only decide for themselves what they should donate to.

My experience is people donate more to libre software rather than closed-source software, and even more so if all opportunities to explore libre software are explored.

Doesn't that sound like a good plan? Why otherwise does the loyalty to the Play store mean it has to be listed first?

File: README.md (hunk 3 above)
Comment (comradekingu) [reply README.md:171]: I don't disagree with your name being listed as the main copyright holder.
I don't disagree with you re-iterating this in every file.
I don't even disagree with your name listed as the copyright holder of every file. However, if it is in a translation you had no part in, that is an error.

If you take any contributions without a CLA, it means copyright is shared between more than one entity.
This is why the GPLv3+ protections mean anything. Nobody can effectively re-license into a weaker license, because everyone (every copyright holder) has to agree.
Nobody has the copyright to files only you wrote btw.

I didn't remove your copyright, I merely added "and contributors". That is accurate.
When I or anyone else contributes it is done as per the GPLv3+ terms.
You and anyone else can use it under those terms, but it doesn't transfer copyright to anyone.

Can you clarify what you think will happen if you merge this 
"Copyright © 2019–2021 Garmatin Oleksandr and contributors" line?

What I think will happen is nothing, and that it is a nice nod to the people that contributed, and that people will be more likely to contribute in the future because of it.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Gedsh) [reply README.md:11]: > Google prevents asking for donations in app store listings.
Google taking a 30% cut of sales is a good reason to list it last.

Google has billions of users looking for apps like InviZible.

> If you make a Liberapay account I can help you promote it and write a pitch for why people should donate.

Thanks. I'll try.

> My experience is people donate more to libre software rather than closed-source software

Privacy and anonymity software should definitely be libre. Otherwise it makes no sense.

> Doesn't that sound like a good plan?

That sounds good. But for now, this is only a plan.

> Why otherwise does the loyalty to the Play store mean it has to be listed first?

Loyalty to users. They provide the opportunity to continue developing the app.

[the remark this answers is not in the document]
Comment (comradekingu) [reply README.md:11]: Seeing how they are already users, and did pick one of the few privacy tools that can make the claim of being transparent because it is libre software, why wouldn't everyone be happy about better choices for privacy, for the same reasons, are promoted?
Try it, see if it works like it does every time, and then I will stop doing it everywhere.
In the greater scheme of things, lets focus on the strings PR, and not what I think the README should be like.
Users, and particularly more new users is what matters.

File: README.md (hunk 2 above)
Comment (Gedsh) [reply README.md:135]: I get it. You are an adherent of modern "freedom". It sounds like this:
You are human. Of course, you have a choice. You have several options, and our best experts will select the best one for you. So you actually have only one option. But it is from experts! It is the best! So make your choice from one option.
Thanks so much. I prefer "classic" freedom with real multiple options.

File: README.md (hunk 3 above)
Comment (Gedsh) [reply README.md:171]: > I don't even disagree with your name listed as the copyright holder of every file. However, if it is in a translation you had no part in, that is an error.

This is the reason I am unable to merge your pull requests.

> Can you clarify what you think will happen if you merge this

InviZible is my child. I feel responsible for its fate. If I add "and ...", it will mean that it is no longer mine, but some of our children. I know it sounds crazy, but I can't do it. So anyone can use, fork or help develop the project. And I appreciate that. But the original InviZible will only be one.

File: README.md (hunk 2 above)
Comment (comradekingu) [reply README.md:135]: <details>
I find it hard to offer a coherent set of options without making the concession that, and i find you agree, some options are less good for good reason. This is the paradox of choice.

Every single libre software project has understood using POEditor is not good PR, bar none? 
The entire ecosystem is not less free because of it.
A further nobody recommends POEditor. Except for this project.
In this project, it appears nobody actually uses POEditor to translate InviZible, but there have been over 400 translations on the non-advertised Weblate platform.

Weblate respects user freedom, on a meaningful level, and it dominates for libre software projects,
without limiting anyone from using other alternatives.

Relying on POEditor in any way means being subject to terms they can change, at any time, and a codebase they can shut down tomorrow. For any appreciation of what freedom is outside of noise, that is important.

Similarly, the GNU GPLv3+ license is not allowing the maximum set of options, it clearly limits bad options to what is a end product that favours user freedom over "I am the user that doesn't want to offer a derivative of the project with the source code"-user.

Having superfluous, excessive, and bad options available contributes to the cost of opportunity.
It creates stress. People have to decide what to do.
I decide I fix my suggestions if there is a good argument, and I assume all other options may be good, but beyond my limit of understanding.

There was one opportunity to use one platform, and now there will be two platforms for no reason, but one will be advertised. Better platform, and more choice.
</details>

Do you think anyone will use POEditor if both are advertised?
Will anyone complain?
Is there any reason to list POEditor rather than the theoretical argument?
In this line of thinking the project should be available and advertised on every platform there is.

File: README.md (hunk 3 above)
Comment (comradekingu) [reply README.md:171]: OK, I get it. What you are thinking of is the name "InviZible". You will be the maintainer of it until you decide not to be, or ultimately die of old age.
Nobody can make the same project under the same name, but they can make a new project starting with the same code-base.

> A “contributor” is a copyright holder who authorizes use under this License of the Program or a work on which the Program is based. The work thus licensed is called the contributor's “contributor version”.

This is from the GPLv3+ license, where it details how contributors grant claims they could have made for patents to everyone that abides by the license.

The copyright isn't for the name, which is the real project, but rather the license to use it under GPLv3+.
Everyone that has a copyright claim, agrees to the GPLv3+ license when contributing.
The license is protected by everyone that has a copyright claim to their parts of the codebase.
If it was all 100% done by you, you could effectively re-license it into a proprietary product.
That can still happen, but now everyone holding a copyright claim to the source code will have to agree.

That you care about the product is all wall and good, and I would definitely think of that as a positive message to send.
What not including "and contributors" does is send the message that you don't care about others.
I can remove "and contributors" if you feel bad about it, but that doesn't actually change anything about the copyright others hold for their parts of the project. Those parts are available to you under GPLv3+, just as your parts are available to others under the same license :)

File: README.md [hunk 6]
```diff
@@ -1,19 +1,24 @@
 # InviZible Pro
 
+Proudly presented by Garmatin Oleksandr, \
```
Comment (comradekingu) [inline README.md:3]: @Gedsh How do you like this paragraph?

File: README.md (hunk 3 above)
Comment (Ilithy) [reply README.md]: > OK, I get it. What you are thinking of is the name "InviZible". You will be the maintainer of it until you decide not to be, or ultimately die of old age. Nobody can make the same project under the same name, but they can make a new project starting with the same code-base.
> 
> > A “contributor” is a copyright holder who authorizes use under this License of the Program or a work on which the Program is based. The work thus licensed is called the contributor's “contributor version”.
> 
> This is from the GPLv3+ license, where it details how contributors grant claims they could have made for patents to everyone that abides by the license.
> 
> The copyright isn't for the name, which is the real project, but rather the license to use it under GPLv3+. Everyone that has a copyright claim, agrees to the GPLv3+ license when contributing. The license is protected by everyone that has a copyright claim to their parts of the codebase. If it was all 100% done by you, you could effectively re-license it into a proprietary product. That can still happen, but now everyone holding a copyright claim to the source code will have to agree.
> 
> That you care about the product is all wall and good, and I would definitely think of that as a positive message to send. What not including "and contributors" does is send the message that you don't care about others. I can remove "and contributors" if you feel bad about it, but that doesn't actually change anything about the copyright others hold for their parts of the project. Those parts are available to you under GPLv3+, just as your parts are available to others under the same license :)

Greetings,
I take the liberty to intervene in this discussion, I personally (and still) made the translation of the application in French (entirely, and I regularly make the translation of the new added elements), 
I feel concerned, because I am a (very small) contributor to the InviZible Pro Project and I approve of the developer of the application in the first place, let me explain:

Demanding a recognition for having worked on a free application, without advertising and developed by a single dev seems to me to be a pettiness, personally if I made this "work" of translation it is above all for the pleasure to offer to the French-speaking users of InviZible Pro the ease of use of a translated application (the French are globally null in English. ..in foreign languages rather^^), not to get my name in the application, moreover, the developer works hard to make this application is quality, it's his job, (and it's a huge coding job behind it) not ours, we must be honest.

Don't see the aggression towards you, it's not the case, from France with love.

File: README.md (hunk 6 above)
Comment (Ilithy) [reply README.md:3]: Hello, this is a mistake, the application is not presented by its developer, but coded, why would you want to reduce the one who launched this project, carries it at arm's length and keeps it alive to a simple standard bearer? It is a nonsense in my opinion

Comment (comradekingu) [reply README.md:3]: OK, I'll replace that with "coded", even though it doesn't flow as nicely in English.

Comment (comradekingu) [inline README.md:3]:
```suggestion
Proudly coded by Garmatin Oleksandr, \
```

Comment (Ilithy) [reply README.md]: > OK, I'll replace that with "coded", even though it doesn't flow as nicely in English.

I agree, what do you think about :
"Proudly manufactured by"

Comment (comradekingu) [reply README.md]: I think "made by" is better.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Sarguncheema) [reply README.md:15]: ###

File: README.md (hunk 3 above)
Comment (comradekingu) [reply README.md]: I isn't petty, but none of the small contributors are doing it (me included).
You don't have your name in the translation (and neither do I).

Read [what I actually put in](https://github.com/Gedsh/InviZible/pull/112/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R3) and you would notice it says the name of the author, first thing.
Nobody else mentioned there.

None taken, your issue is purely with how copyright works.
I can explain copyright or the psychology behind appearing charitable if you want.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (comradekingu) [reply README.md]: @Sarguncheema You didn't leave a comment, but I thought to add
>Google has billions of users looking for apps like InviZible.

Are they looking for it in the README of the GitHub project page?

If users are still here and so clueless as to not know the Play store and F-Droid uses tracking, the first option is the best option. Makes sense to me.

File: README.md (hunk 2 above)
Comment (comradekingu) [reply README.md]: To add to this argument, why not offer the option to route DNS onto Google's server in the app? Terrible freedom, that doesn't add anything. It is just a worse option, like Poeditor is.
Why are users worth less once they are translators contributing to the app?
Why split the community of translators between different ecosystems?
Why should prospect translators have to decide between two platforms and have to find out which one is the best?

Paradox of overloading users with bad choices does not grant more user freedom.
